cli/src/main/kotlin/cli/function/BufferedInterpret.kt

```diff
@@ -24,12 +30,14 @@ class BufferedInterpret(
     private val logger: Logger = ConsoleLogger(),
 ) : CLIFunction {
     override fun run(args: Map<String, String>) {
-        val ast = generateBufferedAST(parser, args)
-
-        val result = StatementInterpreter(getInterpreterMap()).interpret(ast)
-
-        for (log in result.logs) {
-            logger.log(log)
-        }
+        val runner = Runner()
+        runner.run(
+            reader = StatementFileReader(File(getFilePath(args)).inputStream(), getVersion(args)),
+            readInputFunction = StandardInputFunction(),
+            parser = parser,
```

Comment:
Parser I think parser should no longer be passed as an argument, because we need the version to pass it to the constructor.